instrumentation/httpclient/src/main/java/io/opentelemetry/instrumentation/auto/httpclient/HttpHeadersInjectAdapter.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.auto.httpclient;
+
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import java.net.http.HttpRequest;
+
+public class HttpHeadersInjectAdapter implements TextMapPropagator.Setter<HttpRequest> {
+  public static final HttpHeadersInjectAdapter SETTER = new HttpHeadersInjectAdapter();
+
+  @Override
+  public void set(HttpRequest carrier, String key, String value) {
+    // Don't do anything because headers are immutable
```

Comment:
So we don't support context propagation via this http client?

Reply:
context propagation is implemented via `HttpHeadersInstrumentation.java` otherwise tests would fail.
this class is a stub because I have to provide implementation to `HttpClientTracer#getSetter()`

Reply:
added javadoc